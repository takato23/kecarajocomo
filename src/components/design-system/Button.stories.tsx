import type { Meta, StoryObj } from '@storybook/react';
import { ShoppingCart, Heart, Download, ChevronRight, Mic } from 'lucide-react';

import { Button } from './Button';

const meta = {
  title: 'Design System/Button',
  component: Button,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    variant: {
      control: 'select',
      options: ['primary', 'secondary', 'ghost', 'glass', 'fresh', 'warm', 'rich', 'golden'],
      description: 'Visual style variant of the button',
    },
    size: {
      control: 'select',
      options: ['sm', 'md', 'lg', 'xl'],
      description: 'Size of the button',
    },
    loading: {
      control: 'boolean',
      description: 'Shows loading spinner',
    },
    disabled: {
      control: 'boolean',
      description: 'Disables the button',
    },
    fullWidth: {
      control: 'boolean',
      description: 'Makes button full width',
    },
    glow: {
      control: 'boolean',
      description: 'Adds glow effect to button',
    },
  },
} satisfies Meta<typeof Button>;

export default meta;
type Story = StoryObj<typeof meta>;

// Basic variants
export const Primary: Story = {
  args: {
    children: 'Primary Button',
    variant: 'primary',
  },
};

export const Secondary: Story = {
  args: {
    children: 'Secondary Button',
    variant: 'secondary',
  },
};

export const Ghost: Story = {
  args: {
    children: 'Ghost Button',
    variant: 'ghost',
  },
};

export const Glass: Story = {
  args: {
    children: 'Glass Button',
    variant: 'glass',
  },
};

// Food theme variants
export const Fresh: Story = {
  args: {
    children: 'Fresh Button',
    variant: 'fresh',
    glow: true,
  },
};

export const Warm: Story = {
  args: {
    children: 'Warm Button',
    variant: 'warm',
    glow: true,
  },
};

export const Rich: Story = {
  args: {
    children: 'Rich Button',
    variant: 'rich',
    glow: true,
  },
};

export const Golden: Story = {
  args: {
    children: 'Golden Button',
    variant: 'golden',
    glow: true,
  },
};

// Sizes
export const Sizes: Story = {
  render: () => (
    <div className="flex items-center gap-4">
      <Button size="sm">Small</Button>
      <Button size="md">Medium</Button>
      <Button size="lg">Large</Button>
      <Button size="xl">Extra Large</Button>
    </div>
  ),
};

// With Icons
export const WithLeftIcon: Story = {
  args: {
    children: 'Add to Cart',
    leftIcon: <ShoppingCart className="w-4 h-4" />,
    variant: 'primary',
  },
};

export const WithRightIcon: Story = {
  args: {
    children: 'Continue',
    rightIcon: <ChevronRight className="w-4 h-4" />,
    variant: 'primary',
  },
};

export const WithBothIcons: Story = {
  args: {
    children: 'Download Recipe',
    leftIcon: <Download className="w-4 h-4" />,
    rightIcon: <ChevronRight className="w-4 h-4" />,
    variant: 'secondary',
  },
};

// States
export const Loading: Story = {
  args: {
    children: 'Loading...',
    loading: true,
    variant: 'primary',
  },
};

export const Disabled: Story = {
  args: {
    children: 'Disabled Button',
    disabled: true,
    variant: 'primary',
  },
};

// Special Cases
export const FullWidth: Story = {
  args: {
    children: 'Full Width Button',
    fullWidth: true,
    variant: 'primary',
  },
  decorators: [
    (Story) => (
      <div className="w-96">
        <Story />
      </div>
    ),
  ],
};

export const IconOnly: Story = {
  args: {
    children: <Heart className="w-5 h-5" />,
    variant: 'ghost',
    size: 'sm',
    className: 'rounded-full aspect-square',
  },
};

// Real Use Cases
export const VoiceButton: Story = {
  args: {
    children: 'Hablar',
    leftIcon: <Mic className="w-4 h-4" />,
    variant: 'fresh',
    glow: true,
  },
};

export const FavoriteButton: Story = {
  args: {
    children: <Heart className="w-5 h-5" />,
    variant: 'ghost',
    size: 'sm',
    className: 'rounded-full aspect-square hover:text-red-500',
  },
};

// Interactive Demo
export const Interactive: Story = {
  render: () => {
    return (
      <div className="space-y-8">
        <div>
          <h3 className="text-lg font-semibold mb-4">Click to see hover effects</h3>
          <div className="flex flex-wrap gap-4">
            <Button variant="primary" glow>Primary with Glow</Button>
            <Button variant="fresh" glow>Fresh with Glow</Button>
            <Button variant="warm" glow>Warm with Glow</Button>
            <Button variant="glass">Glass Effect</Button>
          </div>
        </div>
        
        <div>
          <h3 className="text-lg font-semibold mb-4">Different states</h3>
          <div className="flex flex-wrap gap-4">
            <Button>Normal</Button>
            <Button loading>Loading</Button>
            <Button disabled>Disabled</Button>
          </div>
        </div>
      </div>
    );
  },
};