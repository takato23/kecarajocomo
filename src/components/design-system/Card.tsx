'use client';

import React from 'react';

import { cn } from '@/lib/utils';

export interface CardProps extends React.HTMLAttributes<HTMLDivElement> {
  variant?: 'default' | 'glass' | 'glass-interactive' | 'fresh' | 'warm' | 'rich' | 'golden';
  padding?: 'none' | 'sm' | 'md' | 'lg' | 'xl';
  hover?: boolean;
  glow?: boolean;
  blur?: boolean;
}

const Card = React.forwardRef<HTMLDivElement, CardProps>(
  (
    {
      className,
      variant = 'default',
      padding = 'md',
      hover = false,
      glow = false,
      blur = true,
      children,
      ...props
    },
    ref
  ) => {
    const baseClasses = [
      'rounded-2xl transition-all duration-300 ease-glass',
      blur && 'backdrop-blur-md',
    ];

    const variantClasses = {
      default: [
        'bg-white border border-neutral-200 shadow-sm',
        'dark:bg-neutral-800 dark:border-neutral-700',
      ],
      glass: [
        'glass',
      ],
      'glass-interactive': [
        'glass-interactive cursor-pointer',
      ],
      fresh: [
        'glass-fresh border border-food-fresh-200/50',
      ],
      warm: [
        'glass-warm border border-food-warm-200/50',
      ],
      rich: [
        'glass-rich border border-food-rich-200/50',
      ],
      golden: [
        'glass-golden border border-food-golden-200/50',
      ],
    };

    const paddingClasses = {
      none: '',
      sm: 'p-3',
      md: 'p-4',
      lg: 'p-6',
      xl: 'p-8',
    };

    const hoverClasses = hover
      ? [
          'hover:transform hover:-translate-y-1 hover:shadow-glass-lg',
          variant === 'glass-interactive' && 'hover:bg-glass-interactive-light',
        ]
      : [];

    const glowClasses = glow
      ? {
          default: 'hover:shadow-lg',
          glass: 'hover:shadow-glass-lg',
          'glass-interactive': 'hover:shadow-glass-lg',
          fresh: 'glow-fresh',
          warm: 'glow-warm',
          rich: 'glow-rich',
          golden: 'glow-golden',
        }[variant]
      : '';

    return (
      <div
        ref={ref}
        className={cn(
          baseClasses,
          variantClasses[variant],
          paddingClasses[padding],
          hoverClasses,
          glowClasses,
          className
        )}
        {...props}
      >
        {children}
      </div>
    );
  }
);

Card.displayName = 'Card';

// Card Header Component
export interface CardHeaderProps extends React.HTMLAttributes<HTMLDivElement> {
  title?: string;
  subtitle?: string;
  action?: React.ReactNode;
}

const CardHeader = React.forwardRef<HTMLDivElement, CardHeaderProps>(
  ({ className, title, subtitle, action, children, ...props }, ref) => {
    return (
      <div
        ref={ref}
        className={cn('flex items-center justify-between mb-4', className)}
        {...props}
      >
        <div className="min-w-0 flex-1">
          {title && (
            <h3 className="text-lg font-semibold text-neutral-900 dark:text-neutral-100 truncate">
              {title}
            </h3>
          )}
          {subtitle && (
            <p className="text-sm text-neutral-500 dark:text-neutral-400 mt-1">
              {subtitle}
            </p>
          )}
          {children}
        </div>
        {action && <div className="ml-4 flex-shrink-0">{action}</div>}
      </div>
    );
  }
);

CardHeader.displayName = 'CardHeader';

// Card Body Component
export interface CardBodyProps extends React.HTMLAttributes<HTMLDivElement> {}

const CardBody = React.forwardRef<HTMLDivElement, CardBodyProps>(
  ({ className, ...props }, ref) => {
    return (
      <div
        ref={ref}
        className={cn('text-neutral-700 dark:text-neutral-300', className)}
        {...props}
      />
    );
  }
);

CardBody.displayName = 'CardBody';

// Card Footer Component
export interface CardFooterProps extends React.HTMLAttributes<HTMLDivElement> {
  justify?: 'start' | 'center' | 'end' | 'between';
}

const CardFooter = React.forwardRef<HTMLDivElement, CardFooterProps>(
  ({ className, justify = 'end', ...props }, ref) => {
    const justifyClasses = {
      start: 'justify-start',
      center: 'justify-center',
      end: 'justify-end',
      between: 'justify-between',
    };

    return (
      <div
        ref={ref}
        className={cn(
          'flex items-center gap-3 mt-4 pt-4 border-t border-neutral-200/50 dark:border-neutral-700/50',
          justifyClasses[justify],
          className
        )}
        {...props}
      />
    );
  }
);

CardFooter.displayName = 'CardFooter';

export { Card, CardHeader, CardBody, CardFooter };