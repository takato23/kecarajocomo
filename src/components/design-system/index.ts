// Base UI Components
export { Button } from './Button';
export type { ButtonProps } from './Button';

export { Input } from './Input';
export type { InputProps } from './Input';

export { Card, CardHeader, CardBody, CardFooter } from './Card';
export type { CardProps, CardHeaderProps, CardBodyProps, CardFooterProps } from './Card';

export { Badge } from './Badge';
export type { BadgeProps } from './Badge';

export { Heading, Text, Code, Link } from './Typography';
export type { HeadingProps, TextProps, CodeProps, LinkProps } from './Typography';

// Specialized Food/Recipe Components
export { RecipeCard } from './RecipeCard';
export type { RecipeCardProps } from './RecipeCard';

export { NutritionDisplay } from './NutritionDisplay';
export type { NutritionDisplayProps, NutritionInfo } from './NutritionDisplay';

// Design System Utilities
export * from './theme';
export * from './icons';