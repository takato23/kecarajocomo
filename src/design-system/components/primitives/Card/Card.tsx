import { forwardRef, HTMLAttributes } from 'react';
import { cva, type VariantProps } from 'class-variance-authority';
import { motion } from 'framer-motion';

import { cn } from '@/lib/utils';

const cardVariants = cva(
  'rounded-xl bg-white overflow-hidden transition-all duration-200',
  {
    variants: {
      variant: {
        default: 'border border-gray-200',
        elevated: 'shadow-lg',
        ghost: 'bg-gray-50',
        glass: 'bg-white/80 backdrop-blur-lg border border-white/20',
      },
      padding: {
        none: 'p-0',
        sm: 'p-3',
        md: 'p-4',
        lg: 'p-6',
        xl: 'p-8',
      },
      interactive: {
        true: 'cursor-pointer hover:scale-[1.02] active:scale-[0.98]',
        false: '',
      },
    },
    defaultVariants: {
      variant: 'default',
      padding: 'md',
      interactive: false,
    },
  }
);

export interface CardProps
  extends HTMLAttributes<HTMLDivElement>,
    VariantProps<typeof cardVariants> {
  asChild?: boolean;
}

export const Card = forwardRef<HTMLDivElement, CardProps>(
  ({ className, variant, padding, interactive, children, onClick, ...props }, ref) => {
    const Component = interactive && onClick ? motion.div : 'div';
    
    const cardProps = interactive && onClick
      ? {
          whileHover: { y: -2 },
          whileTap: { scale: 0.98 },
          transition: { duration: 0.2 },
        }
      : {};
    
    return (
      <Component
        ref={ref}
        className={cn(cardVariants({ variant, padding, interactive, className }))}
        onClick={onClick}
        {...cardProps}
        {...props}
      >
        {children}
      </Component>
    );
  }
);

Card.displayName = 'Card';

// Card Header component
export interface CardHeaderProps extends HTMLAttributes<HTMLDivElement> {
  noPadding?: boolean;
}

export const CardHeader = forwardRef<HTMLDivElement, CardHeaderProps>(
  ({ className, noPadding = false, children, ...props }, ref) => {
    return (
      <div
        ref={ref}
        className={cn(
          'flex flex-col space-y-1.5',
          !noPadding && 'p-6 pb-4',
          className
        )}
        {...props}
      >
        {children}
      </div>
    );
  }
);

CardHeader.displayName = 'CardHeader';

// Card Title component
export interface CardTitleProps extends HTMLAttributes<HTMLHeadingElement> {
  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
}

export const CardTitle = forwardRef<HTMLHeadingElement, CardTitleProps>(
  ({ className, as: Component = 'h3', children, ...props }, ref) => {
    return (
      <Component
        ref={ref}
        className={cn('text-lg font-semibold leading-none tracking-tight', className)}
        {...props}
      >
        {children}
      </Component>
    );
  }
);

CardTitle.displayName = 'CardTitle';

// Card Description component
export const CardDescription = forwardRef<
  HTMLParagraphElement,
  HTMLAttributes<HTMLParagraphElement>
>(({ className, children, ...props }, ref) => {
  return (
    <p
      ref={ref}
      className={cn('text-sm text-gray-500', className)}
      {...props}
    >
      {children}
    </p>
  );
});

CardDescription.displayName = 'CardDescription';

// Card Content component
export interface CardContentProps extends HTMLAttributes<HTMLDivElement> {
  noPadding?: boolean;
}

export const CardContent = forwardRef<HTMLDivElement, CardContentProps>(
  ({ className, noPadding = false, children, ...props }, ref) => {
    return (
      <div
        ref={ref}
        className={cn(!noPadding && 'p-6 pt-0', className)}
        {...props}
      >
        {children}
      </div>
    );
  }
);

CardContent.displayName = 'CardContent';

// Card Footer component
export interface CardFooterProps extends HTMLAttributes<HTMLDivElement> {
  noPadding?: boolean;
}

export const CardFooter = forwardRef<HTMLDivElement, CardFooterProps>(
  ({ className, noPadding = false, children, ...props }, ref) => {
    return (
      <div
        ref={ref}
        className={cn(
          'flex items-center',
          !noPadding && 'p-6 pt-0',
          className
        )}
        {...props}
      >
        {children}
      </div>
    );
  }
);

CardFooter.displayName = 'CardFooter';

// Card Image component
export interface CardImageProps extends HTMLAttributes<HTMLDivElement> {
  src: string;
  alt: string;
  height?: string | number;
  objectFit?: 'cover' | 'contain' | 'fill' | 'none' | 'scale-down';
}

export const CardImage = forwardRef<HTMLDivElement, CardImageProps>(
  ({ className, src, alt, height = 200, objectFit = 'cover', ...props }, ref) => {
    return (
      <div
        ref={ref}
        className={cn('relative overflow-hidden', className)}
        style={{ height: typeof height === 'number' ? `${height}px` : height }}
        {...props}
      >
        <img
          src={src}
          alt={alt}
          className="w-full h-full"
          style={{ objectFit }}
        />
      </div>
    );
  }
);

CardImage.displayName = 'CardImage';