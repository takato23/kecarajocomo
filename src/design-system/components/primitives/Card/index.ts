export {
  Card,
  CardHeader,
  CardTitle,
  CardDescription,
  CardContent,
  CardFooter,
  CardImage,
} from './Card';

export type {
  CardProps,
  CardHeaderProps,
  CardTitleProps,
  CardContentProps,
  CardFooterProps,
  CardImageProps,
} from './Card';