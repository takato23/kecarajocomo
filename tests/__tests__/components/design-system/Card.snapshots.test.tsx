import React from 'react';
import { render } from '@testing-library/react';
import { Card, CardHeader, CardBody, CardFooter } from '@/components/design-system/Card';

// Mock components for consistent snapshots
const MockButton = () => <button>Action</button>;
const MockContent = () => <div>Sample content</div>;

describe('Card Snapshots', () => {
  describe('Card Variants', () => {
    it('should render default card', () => {
      const { container } = render(
        <Card>
          <MockContent />
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render glass card', () => {
      const { container } = render(
        <Card variant="glass">
          <MockContent />
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render glass interactive card', () => {
      const { container } = render(
        <Card variant="glass-interactive">
          <MockContent />
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render fresh card', () => {
      const { container } = render(
        <Card variant="fresh">
          <MockContent />
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render warm card', () => {
      const { container } = render(
        <Card variant="warm">
          <MockContent />
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render rich card', () => {
      const { container } = render(
        <Card variant="rich">
          <MockContent />
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render golden card', () => {
      const { container } = render(
        <Card variant="golden">
          <MockContent />
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Card Padding', () => {
    it('should render card with no padding', () => {
      const { container } = render(
        <Card padding="none">
          <MockContent />
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render card with small padding', () => {
      const { container } = render(
        <Card padding="sm">
          <MockContent />
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render card with medium padding', () => {
      const { container } = render(
        <Card padding="md">
          <MockContent />
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render card with large padding', () => {
      const { container } = render(
        <Card padding="lg">
          <MockContent />
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render card with extra large padding', () => {
      const { container } = render(
        <Card padding="xl">
          <MockContent />
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Card States', () => {
    it('should render card with hover effects', () => {
      const { container } = render(
        <Card hover>
          <MockContent />
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render card with glow effect', () => {
      const { container } = render(
        <Card glow>
          <MockContent />
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render card without blur', () => {
      const { container } = render(
        <Card blur={false}>
          <MockContent />
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('CardHeader Snapshots', () => {
    it('should render card header with title', () => {
      const { container } = render(
        <CardHeader title="Card Title" />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render card header with title and subtitle', () => {
      const { container } = render(
        <CardHeader title="Card Title" subtitle="Card subtitle" />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render card header with action', () => {
      const { container } = render(
        <CardHeader title="Card Title" action={<MockButton />} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render card header with all props', () => {
      const { container } = render(
        <CardHeader 
          title="Card Title" 
          subtitle="Card subtitle" 
          action={<MockButton />} 
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render card header with custom children', () => {
      const { container } = render(
        <CardHeader>
          <div>Custom header content</div>
        </CardHeader>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('CardBody Snapshots', () => {
    it('should render card body with text content', () => {
      const { container } = render(
        <CardBody>
          <p>This is card body content</p>
        </CardBody>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render card body with custom className', () => {
      const { container } = render(
        <CardBody className="custom-body">
          <p>Custom styled body</p>
        </CardBody>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('CardFooter Snapshots', () => {
    it('should render card footer with default justify', () => {
      const { container } = render(
        <CardFooter>
          <MockButton />
        </CardFooter>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render card footer with start justify', () => {
      const { container } = render(
        <CardFooter justify="start">
          <MockButton />
        </CardFooter>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render card footer with center justify', () => {
      const { container } = render(
        <CardFooter justify="center">
          <MockButton />
        </CardFooter>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render card footer with between justify', () => {
      const { container } = render(
        <CardFooter justify="between">
          <MockButton />
          <MockButton />
        </CardFooter>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Complete Card Snapshots', () => {
    it('should render complete card with all components', () => {
      const { container } = render(
        <Card variant="glass" padding="lg" hover glow>
          <CardHeader 
            title="Complete Card" 
            subtitle="This is a complete card example" 
            action={<MockButton />}
          />
          <CardBody>
            <p>This is the main content of the card. It can contain any content.</p>
            <MockContent />
          </CardBody>
          <CardFooter justify="between">
            <MockButton />
            <MockButton />
          </CardFooter>
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render fresh card with simple content', () => {
      const { container } = render(
        <Card variant="fresh" padding="md">
          <CardHeader title="Fresh Card" />
          <CardBody>
            <p>Fresh card content</p>
          </CardBody>
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render interactive card with hover', () => {
      const { container } = render(
        <Card variant="glass-interactive" hover>
          <CardHeader title="Interactive Card" />
          <CardBody>
            <p>Click me!</p>
          </CardBody>
        </Card>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});