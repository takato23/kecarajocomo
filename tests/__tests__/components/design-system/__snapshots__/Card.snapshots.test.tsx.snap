// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Snapshots Card Padding should render card with extra large padding 1`] = `
<div
  class="rounded-2xl transition-all duration-300 ease-glass backdrop-blur-md bg-white border border-neutral-200 shadow-sm dark:bg-neutral-800 dark:border-neutral-700 p-8"
>
  <div>
    Sample content
  </div>
</div>
`;

exports[`Card Snapshots Card Padding should render card with large padding 1`] = `
<div
  class="rounded-2xl transition-all duration-300 ease-glass backdrop-blur-md bg-white border border-neutral-200 shadow-sm dark:bg-neutral-800 dark:border-neutral-700 p-6"
>
  <div>
    Sample content
  </div>
</div>
`;

exports[`Card Snapshots Card Padding should render card with medium padding 1`] = `
<div
  class="rounded-2xl transition-all duration-300 ease-glass backdrop-blur-md bg-white border border-neutral-200 shadow-sm dark:bg-neutral-800 dark:border-neutral-700 p-4"
>
  <div>
    Sample content
  </div>
</div>
`;

exports[`Card Snapshots Card Padding should render card with no padding 1`] = `
<div
  class="rounded-2xl transition-all duration-300 ease-glass backdrop-blur-md bg-white border border-neutral-200 shadow-sm dark:bg-neutral-800 dark:border-neutral-700"
>
  <div>
    Sample content
  </div>
</div>
`;

exports[`Card Snapshots Card Padding should render card with small padding 1`] = `
<div
  class="rounded-2xl transition-all duration-300 ease-glass backdrop-blur-md bg-white border border-neutral-200 shadow-sm dark:bg-neutral-800 dark:border-neutral-700 p-3"
>
  <div>
    Sample content
  </div>
</div>
`;

exports[`Card Snapshots Card States should render card with glow effect 1`] = `
<div
  class="rounded-2xl transition-all duration-300 ease-glass backdrop-blur-md bg-white border border-neutral-200 shadow-sm dark:bg-neutral-800 dark:border-neutral-700 p-4 hover:shadow-lg"
>
  <div>
    Sample content
  </div>
</div>
`;

exports[`Card Snapshots Card States should render card with hover effects 1`] = `
<div
  class="rounded-2xl transition-all duration-300 ease-glass backdrop-blur-md bg-white border border-neutral-200 shadow-sm dark:bg-neutral-800 dark:border-neutral-700 p-4 hover:transform hover:-translate-y-1 hover:shadow-glass-lg"
>
  <div>
    Sample content
  </div>
</div>
`;

exports[`Card Snapshots Card States should render card without blur 1`] = `
<div
  class="rounded-2xl transition-all duration-300 ease-glass bg-white border border-neutral-200 shadow-sm dark:bg-neutral-800 dark:border-neutral-700 p-4"
>
  <div>
    Sample content
  </div>
</div>
`;

exports[`Card Snapshots Card Variants should render default card 1`] = `
<div
  class="rounded-2xl transition-all duration-300 ease-glass backdrop-blur-md bg-white border border-neutral-200 shadow-sm dark:bg-neutral-800 dark:border-neutral-700 p-4"
>
  <div>
    Sample content
  </div>
</div>
`;

exports[`Card Snapshots Card Variants should render fresh card 1`] = `
<div
  class="rounded-2xl transition-all duration-300 ease-glass backdrop-blur-md glass-fresh border border-food-fresh-200/50 p-4"
>
  <div>
    Sample content
  </div>
</div>
`;

exports[`Card Snapshots Card Variants should render glass card 1`] = `
<div
  class="rounded-2xl transition-all duration-300 ease-glass backdrop-blur-md glass p-4"
>
  <div>
    Sample content
  </div>
</div>
`;

exports[`Card Snapshots Card Variants should render glass interactive card 1`] = `
<div
  class="rounded-2xl transition-all duration-300 ease-glass backdrop-blur-md glass-interactive cursor-pointer p-4"
>
  <div>
    Sample content
  </div>
</div>
`;

exports[`Card Snapshots Card Variants should render golden card 1`] = `
<div
  class="rounded-2xl transition-all duration-300 ease-glass backdrop-blur-md glass-golden border border-food-golden-200/50 p-4"
>
  <div>
    Sample content
  </div>
</div>
`;

exports[`Card Snapshots Card Variants should render rich card 1`] = `
<div
  class="rounded-2xl transition-all duration-300 ease-glass backdrop-blur-md glass-rich border border-food-rich-200/50 p-4"
>
  <div>
    Sample content
  </div>
</div>
`;

exports[`Card Snapshots Card Variants should render warm card 1`] = `
<div
  class="rounded-2xl transition-all duration-300 ease-glass backdrop-blur-md glass-warm border border-food-warm-200/50 p-4"
>
  <div>
    Sample content
  </div>
</div>
`;

exports[`Card Snapshots CardBody Snapshots should render card body with custom className 1`] = `
<div
  class="text-neutral-700 dark:text-neutral-300 custom-body"
>
  <p>
    Custom styled body
  </p>
</div>
`;

exports[`Card Snapshots CardBody Snapshots should render card body with text content 1`] = `
<div
  class="text-neutral-700 dark:text-neutral-300"
>
  <p>
    This is card body content
  </p>
</div>
`;

exports[`Card Snapshots CardFooter Snapshots should render card footer with between justify 1`] = `
<div
  class="flex items-center gap-3 mt-4 pt-4 border-t border-neutral-200/50 dark:border-neutral-700/50 justify-between"
>
  <button>
    Action
  </button>
  <button>
    Action
  </button>
</div>
`;

exports[`Card Snapshots CardFooter Snapshots should render card footer with center justify 1`] = `
<div
  class="flex items-center gap-3 mt-4 pt-4 border-t border-neutral-200/50 dark:border-neutral-700/50 justify-center"
>
  <button>
    Action
  </button>
</div>
`;

exports[`Card Snapshots CardFooter Snapshots should render card footer with default justify 1`] = `
<div
  class="flex items-center gap-3 mt-4 pt-4 border-t border-neutral-200/50 dark:border-neutral-700/50 justify-end"
>
  <button>
    Action
  </button>
</div>
`;

exports[`Card Snapshots CardFooter Snapshots should render card footer with start justify 1`] = `
<div
  class="flex items-center gap-3 mt-4 pt-4 border-t border-neutral-200/50 dark:border-neutral-700/50 justify-start"
>
  <button>
    Action
  </button>
</div>
`;

exports[`Card Snapshots CardHeader Snapshots should render card header with action 1`] = `
<div
  class="flex items-center justify-between mb-4"
>
  <div
    class="min-w-0 flex-1"
  >
    <h3
      class="text-lg font-semibold text-neutral-900 dark:text-neutral-100 truncate"
    >
      Card Title
    </h3>
  </div>
  <div
    class="ml-4 flex-shrink-0"
  >
    <button>
      Action
    </button>
  </div>
</div>
`;

exports[`Card Snapshots CardHeader Snapshots should render card header with all props 1`] = `
<div
  class="flex items-center justify-between mb-4"
>
  <div
    class="min-w-0 flex-1"
  >
    <h3
      class="text-lg font-semibold text-neutral-900 dark:text-neutral-100 truncate"
    >
      Card Title
    </h3>
    <p
      class="text-sm text-neutral-500 dark:text-neutral-400 mt-1"
    >
      Card subtitle
    </p>
  </div>
  <div
    class="ml-4 flex-shrink-0"
  >
    <button>
      Action
    </button>
  </div>
</div>
`;

exports[`Card Snapshots CardHeader Snapshots should render card header with custom children 1`] = `
<div
  class="flex items-center justify-between mb-4"
>
  <div
    class="min-w-0 flex-1"
  >
    <div>
      Custom header content
    </div>
  </div>
</div>
`;

exports[`Card Snapshots CardHeader Snapshots should render card header with title 1`] = `
<div
  class="flex items-center justify-between mb-4"
>
  <div
    class="min-w-0 flex-1"
  >
    <h3
      class="text-lg font-semibold text-neutral-900 dark:text-neutral-100 truncate"
    >
      Card Title
    </h3>
  </div>
</div>
`;

exports[`Card Snapshots CardHeader Snapshots should render card header with title and subtitle 1`] = `
<div
  class="flex items-center justify-between mb-4"
>
  <div
    class="min-w-0 flex-1"
  >
    <h3
      class="text-lg font-semibold text-neutral-900 dark:text-neutral-100 truncate"
    >
      Card Title
    </h3>
    <p
      class="text-sm text-neutral-500 dark:text-neutral-400 mt-1"
    >
      Card subtitle
    </p>
  </div>
</div>
`;

exports[`Card Snapshots Complete Card Snapshots should render complete card with all components 1`] = `
<div
  class="rounded-2xl transition-all duration-300 ease-glass backdrop-blur-md glass p-6 hover:transform hover:-translate-y-1 hover:shadow-glass-lg"
>
  <div
    class="flex items-center justify-between mb-4"
  >
    <div
      class="min-w-0 flex-1"
    >
      <h3
        class="text-lg font-semibold text-neutral-900 dark:text-neutral-100 truncate"
      >
        Complete Card
      </h3>
      <p
        class="text-sm text-neutral-500 dark:text-neutral-400 mt-1"
      >
        This is a complete card example
      </p>
    </div>
    <div
      class="ml-4 flex-shrink-0"
    >
      <button>
        Action
      </button>
    </div>
  </div>
  <div
    class="text-neutral-700 dark:text-neutral-300"
  >
    <p>
      This is the main content of the card. It can contain any content.
    </p>
    <div>
      Sample content
    </div>
  </div>
  <div
    class="flex items-center gap-3 mt-4 pt-4 border-t border-neutral-200/50 dark:border-neutral-700/50 justify-between"
  >
    <button>
      Action
    </button>
    <button>
      Action
    </button>
  </div>
</div>
`;

exports[`Card Snapshots Complete Card Snapshots should render fresh card with simple content 1`] = `
<div
  class="rounded-2xl transition-all duration-300 ease-glass backdrop-blur-md glass-fresh border border-food-fresh-200/50 p-4"
>
  <div
    class="flex items-center justify-between mb-4"
  >
    <div
      class="min-w-0 flex-1"
    >
      <h3
        class="text-lg font-semibold text-neutral-900 dark:text-neutral-100 truncate"
      >
        Fresh Card
      </h3>
    </div>
  </div>
  <div
    class="text-neutral-700 dark:text-neutral-300"
  >
    <p>
      Fresh card content
    </p>
  </div>
</div>
`;

exports[`Card Snapshots Complete Card Snapshots should render interactive card with hover 1`] = `
<div
  class="rounded-2xl transition-all duration-300 ease-glass backdrop-blur-md glass-interactive cursor-pointer p-4 hover:transform hover:-translate-y-1 hover:shadow-glass-lg hover:bg-glass-interactive-light"
>
  <div
    class="flex items-center justify-between mb-4"
  >
    <div
      class="min-w-0 flex-1"
    >
      <h3
        class="text-lg font-semibold text-neutral-900 dark:text-neutral-100 truncate"
      >
        Interactive Card
      </h3>
    </div>
  </div>
  <div
    class="text-neutral-700 dark:text-neutral-300"
  >
    <p>
      Click me!
    </p>
  </div>
</div>
`;
